app/src/main/java/com/circle/android/api/OkHttpStack.java

```diff
@@ -0,0 +1,183 @@
+
+import javax.annotation.Nullable;
+
+
+/**
+ * OkHttp backed {@link com.android.volley.toolbox.HttpStack HttpStack} that does not
+ * use okhttp-urlconnection
+ */
+public class OkHttpStack implements HttpStack {
+
+    private final OkHttpClient mClient;
+
+    public OkHttpStack(OkHttpClient client) {
+        this.mClient = client;
+    }
+
+    @Override
+    public HttpResponse performRequest(Request<?> request, Map<String, String> additionalHeaders)
+        throws IOException, AuthFailureError {
+
+        OkHttpClient client = mClient.clone();
+        int timeoutMs = request.getTimeoutMs();
+        client.setConnectTimeout(timeoutMs, TimeUnit.MILLISECONDS);
+        client.setReadTimeout(timeoutMs, TimeUnit.MILLISECONDS);
+        client.setWriteTimeout(timeoutMs, TimeUnit.MILLISECONDS);
+
+        com.squareup.okhttp.Request.Builder okHttpRequestBuilder = new com.squareup.okhttp.Request.Builder();
+        okHttpRequestBuilder.url(request.getUrl());
+
+        Map<String, String> headers = request.getHeaders();
+        for (final String name : headers.keySet()) {
+            okHttpRequestBuilder.addHeader(name, headers.get(name));
+        }
+        for (final String name : additionalHeaders.keySet()) {
+            okHttpRequestBuilder.addHeader(name, additionalHeaders.get(name));
+        }
+
+        setConnectionParametersForRequest(okHttpRequestBuilder, request);
+
+        com.squareup.okhttp.Request okHttpRequest = okHttpRequestBuilder.build();
+        Call okHttpCall = client.newCall(okHttpRequest);
+        Response okHttpResponse = okHttpCall.execute();
+
+        StatusLine responseStatus = new BasicStatusLine(parseProtocol(okHttpResponse.protocol()), okHttpResponse.code(), okHttpResponse.message());
+        BasicHttpResponse response = new BasicHttpResponse(responseStatus);
+        response.setEntity(entityFromOkHttpResponse(okHttpResponse));
+
+        Headers responseHeaders = okHttpResponse.headers();
+        for (int i = 0, len = responseHeaders.size(); i < len; i++) {
+            final String name = responseHeaders.name(i), value = responseHeaders.value(i);
+            if (name != null) {
+                response.addHeader(new BasicHeader(name, value));
+            }
+        }
+
+        return response;
+    }
+
+    private static HttpEntity entityFromOkHttpResponse(Response r) throws IOException {
+        BasicHttpEntity entity = new BasicHttpEntity();
+        ResponseBody body = r.body();
+
+        entity.setContent(body.byteStream());
+        entity.setContentLength(body.contentLength());
+        entity.setContentEncoding(r.header("Content-Encoding"));
+
+        if (body.contentType() != null) {
+            entity.setContentType(body.contentType().type());
+        }
+        return entity;
+    }
+
+    @SuppressWarnings("deprecation")
+    private static void setConnectionParametersForRequest(com.squareup.okhttp.Request.Builder builder, Request<?> request)
```

Comment:
You're right that this doesn't conform to our style guide.  However, this is open source code written by someone else, which is usually better to change as little as possible, in case the original code changes and we want to apply diffs.  (In fact, there were a couple of other small changes I made to the original, such as a spelling correction, but on further thought it's probably best to back them out, so I have undone my changes.)